src/VisualStudio/IntegrationTest/New.IntegrationTests/MefCompositionTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.VisualStudio.IntegrationTests;
+using Xunit;
+
+namespace Roslyn.VisualStudio.NewIntegrationTests
+{
+    public class MefCompositionTests : AbstractIntegrationTest
+    {
+        [IdeFact]
+        public async Task AssertNoCompositionFailures()
+        {
+            // Read the .err file that contains errors; this isn't a great way to do it but we have no
+            // better option at this point.
+            var shell = await TestServices.Shell.GetRequiredGlobalServiceAsync<SVsSettingsManager, IVsSettingsManager>(HangMitigatingCancellationToken);
+            Marshal.ThrowExceptionForHR(shell.GetApplicationDataFolder((uint)__VsApplicationDataFolder.ApplicationDataFolder_LocalSettings, out var applicationDataFolder));
+
+            var compositionErrorFileLines = File.ReadAllLines(Path.Combine(applicationDataFolder, @"ComponentModelCache\Microsoft.VisualStudio.Default.err"));
+            var compositionErrors = new List<string>();
+
+            for (int i = 0; i < compositionErrorFileLines.Length; i++)
+            {
+                var line = compositionErrorFileLines[i];
+
+                if (line.EndsWith("expected exactly 1 export matching constraints:") &&
+                    StartsWithApplicableSymbol(line))
+                {
+                    var entireError = string.Join(Environment.NewLine, compositionErrorFileLines.Skip(i).TakeWhile(s => !string.IsNullOrWhiteSpace(s)));
+                    compositionErrors.Add(entireError);
+                }
+            }
+
+            Assert.Empty(compositionErrors);
```

Comment:
The error message produced here is not usable.

```suggestion
            AssertEx.EqualOrDiff("", string.Join(Environment.NewLine, compositionErrors));
```

Reply:
Updated.